src/main/kotlin/net/atos/client/zgw/shared/ZGWApiService.kt

```diff
@@ -250,36 +249,41 @@ class ZGWApiService @Inject constructor(
      * @param zaak [Zaak]
      * @return [RolOrganisatorischeEenheid] or 'null'.
      */
-    fun findGroepForZaak(zaak: Zaak): Optional<RolOrganisatorischeEenheid> =
-        findBehandelaarRoleForZaak(zaak, BetrokkeneType.ORGANISATORISCHE_EENHEID)
-            .map { RolOrganisatorischeEenheid::class.java.cast(it) }
+    fun findGroepForZaak(zaak: Zaak): RolOrganisatorischeEenheid? =
+        findBehandelaarRoleForZaak(zaak, BetrokkeneType.ORGANISATORISCHE_EENHEID)?.let {
+            RolOrganisatorischeEenheid::class.java.cast(it)
```

Comment:
Is this not how we can cast in Kotlin?
```suggestion
            it as RolOrganisatorischeEenheid
```

Reply:
true! the IntelliJ convert to Kotlin functionality is not as clever as you.. ;-)

Reply:
done